src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        final Node<T> link = last;
+        final Node<T> node = new Node<>(link, value, null);
+        last = node;
+        if (link == null) {
+            first = node;
+        } else {
+            link.next = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            Node<T> nextNode = getNode(index);
+            linkBefore(value, nextNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToUnlink = getNode(index);
+        return unlink(nodeToUnlink);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> temp = first;
+        for (int i = 0; i < size; i++) {
+            if (object == temp.item || object != null && object.equals(temp.item)) {
+                unlink(temp);
+                return true;
+            }
+            temp = temp.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T item;
```

Comment:
```suggestion
        private T value;
```
You have code like `T oldValue = node.item;` So, let's use one word.